### PR TITLE
Don't use ADSB messages with undefined fields in navigator

### PR DESCRIPTION
**Pull request by svpcom**
Ignore messages that have undefined fields required for collision estimation.

### CODE REVIEW

**Comment by dagar** [comment]
@svpcom check CI failure (trivial style issue).

@LorenzMeier do we still need to limit direct mavlink usage to the mavlink module?

**Comment by svpcom** [comment]
@dagar Style fixed

**Comment by dagar** [comment]
In the past we had to be careful to not include mavlink headers outside of the mavlink module. We need to verify if that's still the case.

**Comment by LorenzMeier** [comment]
That's still the case because we don't want to start leaking MAVLink as a dependency into the system. In particular now that FastRTPS is picking up speed.

File: src/modules/navigator/navigator_main.cpp [hunk 1]
```diff
@@ -58,6 +58,7 @@
 #include <px4_defines.h>
 #include <px4_posix.h>
 #include <px4_tasks.h>
+#include <v2.0/common/mavlink.h>
```
Comment (LorenzMeier) [inline src/modules/navigator/navigator_main.cpp]: This needs to be changed as it hardwires MAVLink into PX4 as the only protocol.

File: src/modules/navigator/navigator_main.cpp [hunk 2]
```diff
@@ -963,7 +964,8 @@ void Navigator::fake_traffic(const char *callsign, float distance, float directi
 	strncpy(&tr.callsign[0], callsign, sizeof(tr.callsign));
 	tr.emitter_type = 0; // Type from ADSB_EMITTER_TYPE enum
 	tr.tslc = 2; // Time since last communication in seconds
-	tr.flags = 0; // Flags to indicate various statuses including valid data fields
+	tr.flags = ADSB_FLAGS_VALID_COORDS | ADSB_FLAGS_VALID_HEADING | ADSB_FLAGS_VALID_VELOCITY | \
```
Comment (LorenzMeier) [inline src/modules/navigator/navigator_main.cpp]: These need to be abstracted in the appropriate UORB message spec.

**Comment by dagar** [comment]
In that case @svpcom can you add the relevant fields to transponder report? https://github.com/PX4/Firmware/blob/master/msg/transponder_report.msg#L12

**Comment by svpcom** [comment]
@daga I can add something like ``all_fields_defined`` to UORB message to indicate that ADSB message has all fields need to run collision detection. But I'm unable to remove ``flags`` field because it used for reconstruct ADSB mavlink message to route it via telemetry link (src/modules/mavlink/mavlink_messages.cpp#1342). 

It seems that is needed for directly attached (to FMU via UART) ADSB receiver which emits messages that needed not only for FMU (collision detection in navigator module), but for QGroundControl too (to display airplanes on the map).

**Comment by LorenzMeier** [comment]
@svpcom You can just copy the flags from MAVLink into the uORB message and then add code when the MAVLink message gets assembled.

**Comment by svpcom** [comment]
@LorenzMeier 
So there are four possible implementations:
1. Copy definition of enum ADSB_FLAGS from mavlink headers. This is simplest variant because we dont't need to map mavlink adsb flags to uORB adsb flags. But if their definitions will be changed in future mavlink versions this break navigator code.
2. Add own enum for adsb flags and make a mapping from mavlink to uORB flags in mavlink module (during parsing and reconstruction phases).
3. Add boolean field for each flag to uORB message. This is mostly equal to previous case.
4. Add additional boolean field (use current ``flags`` field only for mavlink packet reconstruction) and set it to true if we have ``ADSB_FLAGS_VALID_COORDS, ADSB_FLAGS_VALID_HEADING, ADSB_FLAGS_VALID_VELOCITY and ADSB_FLAGS_VALID_ALTITUDE`` set in message.  This doesn't required to copy enum ADSB_FLAGS or add additional mapping.

Also the current  implementation always retranslate ADSB mavlink messages. Even if these message were emitted by external receiver and got via telemetry link. I use modified [dump1090](https://github.com/svpcom/dump1090) which run on companion computer and use mavlink-router to inject packets to telemetry stream. I don't have http://www.uavionix.com/products/pingrx/ and don't know which sys_id and comp_id are used in their messages to make a decision - retranslate message to telemetry link or not.

So my questions are: 
1. What implementation to choose?
2. What to do with duplication of adsb messages which are received via telemetry link?

**Comment by LorenzMeier** [comment]
The best is to start with a literal copy but still add the code that translates between them (rather than just copying the value). This is future-proof but right now the least effort. If you already see that you would like to do more medium-term than is supported by MAVLink right now you can already now diverge where needed.

**Comment by LorenzMeier** [comment]
@dagar Please squash if you merge before me.

**Comment by svpcom** [comment]
I've fix a typo.

**Comment by LorenzMeier** [comment]
@svpcom Style check failed.

File: src/modules/navigator/navigator.h [hunk 3]
```diff
@@ -79,6 +79,18 @@
  */
 #define NAVIGATOR_MODE_ARRAY_SIZE 11
 
+typedef enum adsb_transponder_flags_e
+{
+	PX4_ADSB_FLAGS_VALID_COORDS=1, /*  | */
```
Comment (dagar) [inline src/modules/navigator/navigator.h]: Why not define these in the uORB message?

**Pull request merged**